drivers/sensor/icm42688/icm42688.h

```diff
@@ -77,6 +159,107 @@ enum icm42688_accel_odr {
 	ICM42688_ACCEL_ODR_500,
 };
 
+static inline enum icm42688_accel_odr icm42688_accel_hz_to_reg(uint16_t hz)
+{
+	if (hz >= 32000) {
+		return ICM42688_ACCEL_ODR_32000;
+	} else if (hz >= 16000) {
+		return ICM42688_ACCEL_ODR_16000;
+	} else if (hz >= 8000) {
+		return ICM42688_ACCEL_ODR_8000;
+	} else if (hz >= 4000) {
+		return ICM42688_ACCEL_ODR_4000;
+	} else if (hz >= 2000) {
+		return ICM42688_ACCEL_ODR_2000;
+	} else if (hz >= 1000) {
+		return ICM42688_ACCEL_ODR_1000;
+	} else if (hz >= 500) {
+		return ICM42688_ACCEL_ODR_500;
+	} else if (hz >= 200) {
+		return ICM42688_ACCEL_ODR_200;
+	} else if (hz >= 100) {
+		return ICM42688_ACCEL_ODR_100;
+	} else if (hz >= 50) {
+		return ICM42688_ACCEL_ODR_50;
+	} else if (hz >= 25) {
+		return ICM42688_ACCEL_ODR_25;
+	} else if (hz >= 12) {
+		return ICM42688_ACCEL_ODR_12_5;
+	} else if (hz >= 6) {
+		return ICM42688_ACCEL_ODR_6_25;
+	} else if (hz >= 3) {
+		return ICM42688_ACCEL_ODR_3_125;
+	} else {
+		return ICM42688_ACCEL_ODR_1_5625;
+	}
+}
+
+static inline void icm42688_accel_reg_to_hz(enum icm42688_accel_odr odr, struct sensor_value *out)
+{
+	switch (odr) {
+	case ICM42688_ACCEL_ODR_32000:
```

Comment:
I could go either way (let me know what you think). Here's what I saw when I tried this:
1. It was a little odd since `ICM42688_ACCEL_ODR_32000` start off the enum at 1 and not 0 so there's an empty entry
2. This works well for converting enum -> value, but feels wasteful the other way (I tried iterating and a binary search). This makes it a bit odd that the two functions don't look symmetric (IMO).

Reply:
Both approaches are equally readable.  I'm okay with using the switch statement.